### ISSUE TITLE
Add support for SAP HANA SMALLDECIMAL data types

### ISSUE THREAD

**Issue by o-shevchenko** [jOOQ/jOOQ#16975]
### Expected behavior

SMALLDECIMAL is represented as Float

### Actual behavior

SAP HANA SMALLDECIMAL(16, 16) is represented as Long during read.
It leads to incorrect number representation
<img width="529" alt="image" src="https://github.com/user-attachments/assets/62d3020c-9a96-41d1-870e-1040ada43d88">
<img width="209" alt="image" src="https://github.com/user-attachments/assets/f037d28a-a064-46bf-a55f-1e7b6dc1e526">

<img width="545" alt="image" src="https://github.com/user-attachments/assets/f082f4eb-184f-4353-9e12-b07789d83168">

### Steps to reproduce the problem

Read data from SAP HANA with SMALLDECIMAL field:
```Kotlin
context.resultQuery("SELECT * FROM table")
```

### jOOQ Version

3.19.10

### Database product and version

SAP HANA

### Java Version

17

### JDBC / R2DBC driver name and version (include name if unofficial driver)

com.sap.cloud.db.jdbc:ngdbc:2.21.11

**Comment by lukaseder** [comment]
Thanks for your report. We can look into supporting the `SMALLDECIMAL` type in the future.

**Comment by o-shevchenko** [comment]
>In the case of precision = 16, scale = 0, I don't think Long is a bad representation for the value.

Sorry for the typo, it's SMALLDECIMAL(16, 16)
<img width="532" alt="image" src="https://github.com/user-attachments/assets/a3683633-4ee9-4f01-aeb9-c828b2f3b165">
And it represented as Long, so we lost part of the value
<img width="200" alt="image" src="https://github.com/user-attachments/assets/55a81688-5d31-4ed9-82c1-779cebefdf1e">
![image](https://github.com/user-attachments/assets/05563ce8-fd1c-4c5c-be24-19b4b7721328)

**Comment by lukaseder** [comment]
The workaround will be the same as I mentioned here. Use `ResultQuery.coerce()`:

https://github.com/jOOQ/jOOQ/issues/16976#issuecomment-2225765322

**Comment by lukaseder** [comment]
This seems to be just HANA's interpretation of the SQL:2016 standard `DECFLOAT` type, which we'll support as well:

- https://github.com/jOOQ/jOOQ/issues/10880

**Comment by lukaseder** [comment]
Closing this as a duplicate of https://github.com/jOOQ/jOOQ/issues/10880

**Comment by lukaseder** [comment]
I have bad news regarding this data type when used in the way that is mentioned here:

```java
ResultSetMetaData d =
connection
    .createStatement()
    .executeQuery("select cast(1 as smalldecimal) from sys.dummy")
    .getMetaData();
System.out.println(d.getColumnTypeName(1));
System.out.println(d.getPrecision(1));
System.out.println(d.getScale(1));
```

This produces

```
DECIMAL
1
0
```

A `DECIMAL(1)` type isn't really what we were intending to do! Even with this table:

```sql
CREATE TABLE t (t smalldecimal);
```

Querying:

```java
ResultSetMetaData d =
connection
    .createStatement()
    .executeQuery("select t from t")
    .getMetaData();
System.out.println(d.getColumnTypeName(1));
System.out.println(d.getPrecision(1));
System.out.println(d.getScale(1));
```

We're getting `DECIMAL(16)`:

```
DECIMAL
16
0
```

Which isn't the `DECFLOAT` style `SMALLDECIMAL` type. jOOQ cannot fix this when the SAP JDBC driver doesn't report the correct type, even with the `DECFLOAT` support being implemented in #10880, so I guess you'll still have to handle this yourself in some way, probably by proxying JDBC as I suggested elsewhere.

You could try reporting this to SAP. It seems to be a bug, in my opinion.
